cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
I find it odd to dereference these pointer to pass them into the `Sql` query classes c'tors... just a matter of taste.